tests/EventEmitterTest.php

```diff
@@ -0,0 +1,288 @@
+        $this->paymentMethodId = $this->faker->paymentMethodId();
+        $this->transactionId = $this->faker->transactionId();
+        $this->transactionReference = $this->faker->transactionReference();
+        $this->riskScore = $this->faker->riskScore();
+
+        $this->customHttpClient = new Client('', array('redirect.disable' => true));
+        $this->eventDispatcher = $this->customHttpClient->getEventDispatcher();
+
+        $this->testSubscriber = new TestSubscriber();
+        $this->eventDispatcher->addSubscriber($this->testSubscriber);
+    }
+
+    /**
+     * Ensures that 'Request' and 'Response' events are emitted when issuing a request.
+     *
+     * @return void
+     */
+    public function testAuthorizeRequestSuccessfulResponseEmitted()
+    {
+        $this->setMockSoapResponse('AuthorizeSuccess.xml', array(
+            'CURRENCY' => $this->currency,
+            'AMOUNT' => $this->amount,
+            'CUSTOMER_ID' => $this->customerId,
+            'CARD_FIRST_SIX' => substr($this->card['number'], 0, 6),
+            'CARD_LAST_FOUR' => substr($this->card['number'], -4),
+            'EXPIRY_MONTH' => $this->card['expiryMonth'],
+            'EXPIRY_YEAR' => $this->card['expiryYear'],
+            'CVV' => $this->card['cvv'],
+            'PAYMENT_METHOD_ID' => $this->paymentMethodId,
+            'TRANSACTION_ID' => $this->transactionId,
+            'TRANSACTION_REFERENCE' => $this->transactionReference,
+            'RISK_SCORE' => $this->riskScore
+        ));
+
+        $class = $this;
+        $this->eventDispatcher
+            ->addListener(
+                Constants::OMNIPAY_REQUEST_BEFORE_SEND,
+                /** @return void */
+                function (RequestEvent $event) use ($class) {
+                    $request = $event['request'];
+                    $class->assertInstanceOf('\Omnipay\Vindicia\Message\AuthorizeRequest', $request);
+                }
+            );
+
+        $this->eventDispatcher
+            ->addListener(
+                Constants::OMNIPAY_RESPONSE_SUCCESS,
+                /** @return void */
+                function (ResponseEvent $event) use ($class) {
+                    $response = $event['response'];
+                    $class->assertInstanceOf('\Omnipay\Vindicia\Message\Response', $response);
+                }
+            );
+
+        $gateway = new Gateway($this->customHttpClient, $this->getHttpRequest());
+        $gateway->setTestMode(true);
+        $request = $gateway->authorize(
+            array(
+                'amount' => $this->amount,
+                'currency' => $this->currency,
+                'card' => $this->card,
+                'customerId' => $this->customerId
+            )
+        );
+        $response = $request->send();
+        $this->assertTrue($response->isSuccessful());
+
+        $eventsDispatched = $this->testSubscriber->eventsDispatched;
+        $this->assertEquals(1, $eventsDispatched['omnipay.request.before_send']);
```

Comment:
any reason you're not using the consts for this?

Reply:
Nope, added!